instrumentation/que/lib/opentelemetry/instrumentation/que/patches/que_job.rb

```diff
@@ -61,6 +62,12 @@ def enqueue(*args, job_options: {}, **arg_opts)
               end
             end
 
+            def bulk_job_options(tags)
+              Thread.current[:que_jobs_to_bulk_insert][:job_options].merge(tags: tags) do |_, a, b|
+                a.is_a?(Array) && b.is_a?(Array) ? a + b : b
```

Comment:
If I recall correctly plus will allocate a new array object. Please consider using  [concat](https://docs.ruby-lang.org/en/3.2/Array.html#method-i-concat), which will append the elements from one array to an existing array:

```suggestion
                a.is_a?(Array) && b.is_a?(Array) ? a.concat( 
b) : b
```